### PR TITLE
fix(core): codify shell restore exit matrix

### PR DESCRIPTION
**Pull request by Sora-bluesky**
## Summary

- Add a core shell lifecycle matrix that fixes default shell, warm-pane, restart, and exit return-path decisions in one place.
- Route existing pane spawn, pane reap, alt-screen crash restore, warm-pane reset, and exit-empty checks through the matrix helpers.
- Add Rust tests for default shell routing, remain-on-exit return paths, abnormal alt-screen restore, exit-empty behavior, and default-shell restart handling.

## Surface Classification

- [x] Public product surface
- [x] Runtime contract surface
- [ ] Contributor/test surface
- [ ] Generated/runtime artifact not tracked
- [ ] Not sure; reviewer help requested

Reference: Repository surface policy.

## Responsibility And Cutover

- Replaced behavior, workflow, config path, or responsibility: scattered shell restore and exit condition checks are centralized behind core lifecycle helpers.
- Expected outcome: default shells, explicit commands, cwd overrides, restart after default-shell changes, and normal/abnormal exits keep their existing behavior but are covered by a single tested matrix.
- Source of truth: core lifecycle code and Rust tests.
- Old paths preserved, redirected, deprecated, or removed: existing CLI options, command names, runtime paths, and launch flows are preserved.

## Verification

- `cargo fmt --check`
- `cargo test --lib shell_lifecycle`
- `cargo test --lib tree::`
- `cargo check`
- `cargo test --lib`
- `git diff --check`
- `pwsh -NoProfile -File scripts\audit-public-surface.ps1`
- `pwsh -NoProfile -File scripts\git-guard.ps1 -Mode full`

## Security And Privacy

- [x] No secrets, credentials, private local paths, browser profile paths, account IDs, customer data, or raw private logs are included.
- [x] Security issues are reported through `SECURITY.md`, not this public pull request.
- [x] Rollback is clear: revert the shell lifecycle helper and call-site refactor commit.

### CODE REVIEW

**Comment by Sora-bluesky** [comment]
@codex review

File: core/src/pane.rs [hunk 1]
```diff
@@ -68,10 +69,19 @@ fn default_shell_name(command: Option<&str>, configured_shell: Option<&str>) ->
 }
 
 pub fn create_window(pty_system: &dyn portable_pty::PtySystem, app: &mut AppState, command: Option<&str>, start_dir: Option<&str>) -> io::Result<()> {
+    // Expand format variables like #{pane_current_path} at spawn time (#111).
+    let expanded_shell = crate::format::expand_format(&app.default_shell, app);
```
[the remark this answers is not in the document]
Comment (Sora-bluesky) [reply core/src/pane.rs]: Fixed in `9eb4c48`: warm transplant is now checked before `expand_format`, so `create_window` and `split_active_with_command` only expand `default-shell` after the warm route is rejected. I also added matrix coverage for the warm route and reran `cargo test --lib shell_lifecycle`, `cargo test --lib`, `audit-public-surface`, and `git-guard`.

**Comment by Sora-bluesky** [comment]
@codex review
